# David Alejandro Velázquez Valdéz A01632648

# Activity: work with the iris dataset

1. Load the iris.csv file in your computer and compare with the data printed above.


2. How many observations (rows) are in total?


3. How many variables (columns) are in total? What do they represent?


4. How many observatoins are for each type of flower? 


5. What is the type of data for each variable?


6. What are the units of each variable?


In [1]:
# Import the packages that we will be using
import matplotlib.pyplot as plt
import pandas as pd                 # For data handling

In [2]:
# Dataset url
urlIris = "datasets/iris/iris.csv"

# Load the dataset
datasetIris = pd.read_csv(urlIris)


In [3]:
datasetIris

sepal length  sepal width  petal length  petal width           class
0             5.1          3.5           1.4          0.2     Iris-setosa
1             4.9          3.0           1.4          0.2     Iris-setosa
2             4.7          3.2           1.3          0.2     Iris-setosa
3             4.6          3.1           1.5          0.2     Iris-setosa
4             5.0          3.6           1.4          0.2     Iris-setosa
..            ...          ...           ...          ...             ...
145           6.7          3.0           5.2          2.3  Iris-virginica
146           6.3          2.5           5.0          1.9  Iris-virginica
147           6.5          3.0           5.2          2.0  Iris-virginica
148           6.2          3.4           5.4          2.3  Iris-virginica
149           5.9          3.0           5.1          1.8  Iris-virginica

[150 rows x 5 columns]

Pregunta 1: Son otros tipos de datos los que se estan analizando, en este caso son flores las que analizamos. 

Pregunta 2: Hay 150 observaciones (rows) en total

Pregunta 3: Hay 5 variables (columns) en total, representan los distintos datos que conjuntan una observación, teniendo información o datos importantes de una observación en específico y sus respectivos números y categorización que representa a cada planta.
Segun el blog iris.names es: 
   1. sepal length in cm
   2. sepal width in cm
   3. petal length in cm
   4. petal width in cm
   5. class: 
      -- Iris Setosa
      -- Iris Versicolour
      -- Iris Virginica

las cuales son las representaciones de las columnas y sus datos

Pregunta 4: Son 50 observaciones por cada tipo de flor 

Pregunta 5: Según las variables obtenidas serían (representadas despues de la flecha): 
   1. sepal length in cm -> float
   2. sepal width in cm  -> float
   3. petal length in cm -> float
   4. petal width in cm  -> float
   5. class:             -> string
   
      -- Iris Setosa
      -- Iris Versicolour
      -- Iris Virginica

Pregunta 6: Según las variables obtenidas serían (representadas despues de la flecha): 
   1. sepal length in cm -> cm
   2. sepal width in cm  -> cm
   3. petal length in cm -> cm
   4. petal width in cm  -> cm
   5. class:             -> string - clasificación
   
      -- Iris Setosa
      -- Iris Versicolour
      -- Iris Virginica
